Library/OcCryptoLib/BigNumLib.h

```diff
@@ -0,0 +1,126 @@
+/**
+  This library performs arbitrary precision arithmetic operations.
+  For more details, please refer to the source files and function headers.
+
+  Copyright (C) 2019, Download-Fritz. All rights reserved.
+
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#ifndef BIG_NUM_LIB_H
+#define BIG_NUM_LIB_H
+
+//
+// Primitives
+//
+
+/**
+  Assigns A the value 0.
+
+  @param[in,out] A  The number to assign to.
+
+**/
+VOID
+BigNumAssign0 (
+  IN OUT OC_BN  *A
+  );
+
+/**
+  Parses a data array into a number. The buffer size must be a multiple of the
+  Word size. The length of Result must precisely fit the required size.
+
+  @param[in,out] Result      The buffer to store the result in.
+  @param[in]     Buffer      The buffer to parse.
+  @param[in]     BufferSize  The size, in bytes, of Buffer.
+  @param[in]     NumWords    The number of Words of Result.
+
+**/
+VOID
+BigNumDataParseBuffer (
+  IN OUT OC_BN_WORD       *Result,
+  IN     CONST UINT8      *Buffer,
+  IN     UINTN            BufferSize,
+  IN     OC_BN_NUM_WORDS  NumWords
+  );
+
+/**
+  Parses a data array into a number. The buffer size must be a multiple of the
+  Word size. The length of Result must precisely fit the required size.
+
+  @param[in,out] Result      The buffer to store the result in.
+  @param[in]     Buffer      The buffer to parse.
+  @param[in]     BufferSize  The size, in bytes, of Buffer.
+
+**/
+VOID
+BigNumParseBuffer (
+  IN OUT OC_BN        *Result,
+  IN     CONST UINT8  *Buffer,
+  IN     UINTN        BufferSize
+  );
+
+/**
+  Swaps the byte order of Word.
+
+  @param[in] Word  The Word to swap.
+
+  @returns  The byte-swapped value of Word.
+
+**/
+OC_BN_WORD
+BigNumSwapWord (
+  IN OC_BN_WORD  Word
```

Comment:
Once again, why does not this take an array of words with a number of words (with in-place or through extra pointer argument swapping)? That would kind of make sense to me, judging from the interfaces above.

Reply:
This function was designed against internal requirements, if you check the only call to it, you'll see it is not a OC_BN operand. The size component could be taken in to swap OC_BN_WORD * to UINT8 * (because latter is not a valid BN anymore due to order), but that sounds more complicated than what is there right now.

Reply:
Right, ok.